util/network-devp2p/src/connection.rs

```diff
@@ -220,14 +220,13 @@ impl Connection {
 
 	/// Register this connection with the IO event loop.
 	pub fn register_socket<Host: Handler>(&self, reg: Token, event_loop: &mut EventLoop<Host>) -> io::Result<()> {
-		if self.registered.load(AtomicOrdering::SeqCst) {
+		if self.registered.compare_and_swap(false, true, AtomicOrdering::SeqCst) {
 			return Ok(());
         }
 		trace!(target: "network", "connection register; token={:?}", reg);
 		if let Err(e) = event_loop.register(&self.socket, reg, self.interest, PollOpt::edge() /* | PollOpt::oneshot() */) { // TODO: oneshot is broken on windows
 			trace!(target: "network", "Failed to register {:?}, {:?}", reg, e);
```

Comment:
I wonder why we don't return here (and set `self.registered` back to `false`).

Reply:
I wonder the same, @arkpar https://github.com/paritytech/parity-ethereum/commit/cfb8671#diff-c378e114f762560f980403b3e66392dcL140-L142?

Reply:
@arkpar I'll go ahead and merge this now, but I'm still curious. :)